c/stcbackground.c

```diff
@@ -0,0 +1,98 @@
+#include "logging.h"
+#include "stcbackground.h"
+
+#define STC_BG_ENTRIES 101
+#define STC_BG_INTERVAL 10
+
+
+static int processInterval(STCBase *stcBase, STCModule *module, int selectStatus) {
+  STCCallbackList* moduleData = (STCCallbackList*)(module->data);
+  STCIntervalCallbackData* callbackList = moduleData->callbackList;
+
+  for (int i = 0; i < STC_BG_ENTRIES; i++) {
+    STCIntervalCallbackData* t = &callbackList[i];
+    if (t->id != -1)  /* initialise */ {
+      
+      // giving ability to disable job by making timeInterval less than zero
+      // as we return callbackdata, user can make timeInterval zero
+      if(t->timeInterval<0) continue;
+
+      t->countInterval+=STC_BG_INTERVAL;
```

Comment:
The STC base can have a variable timeout (see the `selectTimeoutMillis` argument in `stcBaseMainLoop()`)? How does this code take this into account? I don't think `STC_BG_INTERVAL` will work.

Reply:
yes, you are right, we dont have have way to configure this value, changed name, hope it helps

```
#define MAIN_WAIT_MILLIS 10000
#define STC_BG_MIN_INTERVAL_SECS MAIN_WAIT_MILLIS/1000
```

Reply:
Again, your value is not tied to `selectTimeoutMillis` in any way. This will not work correctly.

Reply:
@ifakhrutdinov  should look good now, i am using time of last run to compute, whether to run again or not, thanks!